hdmf_ml/results_table.py

```diff
@@ -0,0 +1,178 @@
+from hdmf.utils import docval, popargs
+from hdmf.backends.hdf5 import H5DataIO
+from hdmf.common import get_class, register_class, DynamicTable, ElementIdentifiers
+from hdmf.container import Container
+import numpy as np
+from sklearn.preprocessing import LabelEncoder
+
+
+data_type = ('array_data', 'data')
+
+
+@register_class('ResultsTable', 'hdmf-ml')
+class ResultsTable(get_class('ResultsTable', 'hdmf-ml')):
+    # override the auto-generated ResultsTable class
+
+    @docval({'name': 'name',        'type': str,          'default': 'root',
+             'doc': 'a name for these results e.g. params1'},
+            {'name': 'description', 'type': str,          'default': 'no description',
+             'doc': 'a human-readable description of the results e.g. training tricks, parameter, etc.'},
+            {'name': 'n_samples', 'type': int, 'doc': 'the number of samples in this table', 'default': None},
+            allow_extra=True)
+    def __init__(self, **kwargs):
+        n_samples = popargs('n_samples', kwargs)
+        super().__init__(**kwargs)
+        self.__n_samples = n_samples
+
+    @property
+    def n_samples(self):
+        return self.__n_samples
+
+    @docval({'name': 'cls',         'type': (str, type), 'doc': 'class for this column'},
+            {'name': 'data',        'type': data_type,   'doc': 'data for this column'},
+            {'name': 'name',        'type': str,         'doc': 'the name of this column'},
+            {'name': 'description', 'type': str,         'doc': 'a description for this column'},
+            {'name': 'dim2',        'type': str,         'doc': 'the argument holding the second dimension', 'default': None},
```

Comment:
Below, dim2 is tested to be an int, list, tuple, or array, but the docval says it needs to be a string. Should the docval be updated?

Reply:
Also please update the doc in this docval to be more descriptive, e.g., the length of the second dimension or second to nth dimension.